### PR TITLE
Move from exceptions to tag clearing

### PR DESCRIPTION
**Pull request by PeterRugg**
As previously discussed at length, move towards tag clearing on non-monotonic register-register operations in RISC-V and suggest it more strongly in general.

Currently this is just an initial cut at some text changes, which should likely be merged alongside the sail changes (https://github.com/CTSRD-CHERI/sail-cheri-riscv/pull/56) once validated in hardware.

### CODE REVIEW

File: chap-model.tex
Comment (francislaus) [inline chap-model.tex:1119]: duplicate word "speculative"

File: chap-model.tex
Comment (arichardson) [inline chap-model.tex:1108]: Maybe something more detailed like this:

```suggestion
For example, this prevents a bounds restrictions performed within a loop from
being hoisted outside the loop, unless that instructions is always executed.
If the loop is not always entered, this could turn a conditional execution of a  trapping instruction into an unconditional one.\footnote{%
This is not just a theoretical possibility -- we observed this happening in the FreeBSD kernel and had to modify the compiler to avoid hoisting any
potentially-trapping CHERI instructions.
}
```

**Review by arichardson — APPROVED** [review]
LGTM once the sail changes have been merged.

**Comment by rwatson** [comment]
Per comments on https://github.com/CTSRD-CHERI/sail-cheri-riscv/pull/56, you can likely GC some exception cause definitions.

**Comment by tariqkurd-repo** [comment]
This PR has been open since May, when do you expect it to be merged? I believe that you're waiting for the SAIL model to be updated to the new spec.

**Comment by PeterRugg** [comment]
@tariqkurd-repo Soon! The sail PR is ready to go (https://github.com/CTSRD-CHERI/sail-cheri-riscv/pull/56). I've been doing all the implementations (Flute is done, just finishing Toooba). Once I confirm that modified Toooba boots CheriBSD, I think we can do the switch. Hopefully this week.

**Comment by tariqkurd-repo** [comment]
That's great, thanks @PeterRugg . I look forwarding to seeing the updated spec.

**Comment by PeterRugg** [comment]
> I do think that the approach we should take is to only document tag clearing in the model chapter and instead add a separate section over in the rationale chapter that discusses the other approach and the reasons we moved away from it. In particular, I think we want the initial architecture-neutral chapters to be as lean as possible so that someone new to CHERI is just reading what the current spec is. If there are A vs B questions that are still open we should keep them in the initial chapters, but when we've made a change going forward (such as split vs merged register files) we should just document what we now do and describe the other approach in the rationale chapter.

Yep, I'm doing a pass now finding all references to it. I'll perhaps be a bit more aggressive in pruning/moving trade-off text.

**Review by arichardson — APPROVED** [review]

**Comment by PeterRugg** [comment]
Okay, I finally finished a pass trying to find all the text that mentioned exceptions on guarded manipulation failures. I could well have missed some though.

File: chap-architecture.tex
Comment (bsdjhb) [inline chap-architecture.tex:3230]: Thanks :)

**Review by bsdjhb — APPROVED** [review]
This generally looks good to me.

**Comment by bsdjhb** [comment]
BTW, as a followup change after this is merged, I would propose that we make the CHERI exception code tables arch-specific (Morello already follows its own thing with separate exceptions rather than an exception code).  For x86 I will likely renumber the values into a single dense encoding omitting the RISC-V specific unaligned base exception.  I suspect for RISC-V we might also want to consider renumbering, or at the least that any variant adopted upstream by RISC-V probably should have renumbered values?

**Comment by PeterRugg** [comment]
Yes, it did strike me as odd that these were architecture-neutral.

**Comment by bsdjhb** [comment]
Still looks good to me.  The x86 chapter has a reference about how it tag clears unlike RISC-V that you might want to update as well (or I can take care of that after this lands if you prefer)

**Review by arichardson — APPROVED** [review]

**Pull request merged**
